components/logins/ios/Logins/LoginsStorage.swift

```diff
@@ -205,6 +205,17 @@ open class LoginsStorage {
         }
     }
 
+    /// Ensure that the record is valid and a duplicate record doesn't exist.
+    open func ensureValid(login: LoginRecord) throws {
```

Comment:
I think this will be a pretty awkward API to use from swift, but I guess they can ask us to improve it if needed.

Reply:
@thomcc What changes do you suggest? I don't  want to delay this work any longer but I also don't want to merge these changes if they're not worthwhile.